pkg/ccl/backupccl/restore_job.go

```diff
@@ -206,6 +210,14 @@ func restoreWithRetry(
 		}
 
 		log.Warningf(restoreCtx, "encountered retryable error: %+v", err)
+		currentPersistedSpans = resumer.job.Progress().Details.(*jobspb.Progress_Restore).Restore.Checkpoint
```

Comment:
For a moment I wondered whether `frontier.Entries` is guaranteed to have stable output wrt how it merges spans.  However, I _think_ we are fine even if it doesn't because the progress tracker that updates the Checkpoint is only going to update it in response to a completed span.